apps/designer/app/canvas/features/text-editor/text-editor.tsx

```diff
@@ -39,6 +49,61 @@ const AutofocusPlugin = () => {
   return null;
 };
 
+const RemoveParagaphsPlugin = () => {
+  const [editor] = useLexicalComposerContext();
+
+  // register own commands before RichTextPlugin
+  // to stop propagation
+  useLayoutEffect(() => {
+    const removeCommand = editor.registerCommand(
+      KEY_ENTER_COMMAND,
+      (event) => {
+        const selection = $getSelection();
+        if (!$isRangeSelection(selection)) {
+          return false;
+        }
+        event?.preventDefault();
+        // returns true which stops propagation
+        return editor.dispatchCommand(INSERT_LINE_BREAK_COMMAND, false);
+      },
+      COMMAND_PRIORITY_EDITOR
+    );
+
+    // flatten pasted paragraphs
+    const removeNodeTransform = editor.registerNodeTransform(
+      RootNode,
+      (node) => {
+        // merge paragraphs into first with line breaks between
+        if (node.getChildrenSize() > 1) {
+          const children = node.getChildren();
+          let first;
+          for (let i = 0; i < children.length; i += 1) {
+            const p = children[i];
```

Comment:
please use full names: paragraph, index

Reply:
it's readable here, but it won't be somewhere else, so lets set a standard

Reply:
also please add an eslint plugin for this https://github.com/webstudio-is/webstudio-designer/issues/397